packages/query/src/block-processor.ts

```diff
@@ -169,16 +170,17 @@ export class BlockProcessor implements BlockProcessorInterface {
         await this.indexedDb.saveFmdParams(compactBlock.fmdParameters);
       }
       if (compactBlock.gasPrices) {
-        await this.indexedDb.saveGasPrices(
-          new GasPrices({
-            assetId: this.stakingAssetId,
-            ...compactBlock.gasPrices,
-          }),
-        );
+        await this.indexedDb.saveGasPrices({
+          ...toPlainMessage(compactBlock.gasPrices),
+          assetId: toPlainMessage(this.stakingAssetId),
+        });
       }
       if (compactBlock.altGasPrices.length) {
-        for (const gasPrice of compactBlock.altGasPrices) {
-          await this.indexedDb.saveGasPrices(gasPrice);
+        for (const altGas of compactBlock.altGasPrices) {
+          await this.indexedDb.saveGasPrices({
+            ...toPlainMessage(altGas),
+            assetId: getAssetIdFromGasPrices(altGas),
+          });
```

Comment:
we need an additional conditional check before spreading `compactBlock.gasPrices `, something like

```
const altGasPrices = {
  ...toPlainMessage(altGas),
  assetId: getAssetIdFromGasPrices(altGas),
};

if (altGasPrices.assetId) {
  await this.indexedDb.saveGasPrices(altGasPrices);
} else {
  continue syncing without error
}
```

Reply:
Yep, you're right. Updated my example above.